busines_logic/calculate_averages.py

```diff
@@ -1,15 +1,16 @@
+from busines_logic.student import Student
+
+
 def prepare_statistics(lines: list[dict[str, list[float]]]) -> list[dict[str, any]]:
     results = []
     for line in lines:
         username = line['username']
         scores = line['scores']
-        average = round(calculate_average(scores), 1)
-        results.append({'username': username, 'scores': scores, 'average': average})
+        # average = round(calculate_average(scores), 1)
+        student = Student(username, scores)
```

Comment:
The goal of having `Student` class is to use it from the beginning, i.e. in the moment when data is read from the input file. This means that `read_scores` function should return `list[Student]` and all other functions should utilise this class instead of raw `dict`.

And `Student` class is (will be) used both by adapters (e.g. `read_scores`) and business logic. Therefore its definition should be put in `ports` directory, according to hexagonal architecture (I will introduce it to you during our call)